src/rust/ide/src/controller/module.rs

```diff
@@ -124,13 +120,10 @@ shared! { Handle
         /// Updates AST after code change.
         pub fn apply_code_change(&mut self,change:&TextChange) -> FallibleResult<()> {
             let mut code         = self.code();
-            let replaced_size    = change.replaced.end - change.replaced.start;
-            let replaced_span    = Span::new(change.replaced.start,replaced_size);
             let replaced_indices = change.replaced.start.value..change.replaced.end.value;
 
             code.replace_range(replaced_indices,&change.inserted);
-            apply_code_change_to_id_map(&mut self.id_map,&replaced_span,&change.inserted);
```

Comment:
We cannot remove this call. After updating code, the old id map (generated from old ast) should be updated using this function, lest all the node ids can become messy.